src/main/java/org/symphonyoss/symphony/messageml/elements/Element.java

```diff
@@ -103,26 +104,52 @@ public Boolean hasIdAttribute() {
   public void buildAll(MessageMLParser parser, org.w3c.dom.Element element) throws InvalidInputException,
       ProcessingException {
     NamedNodeMap attr = element.getAttributes();
-
     for (int i = 0; i < attr.getLength(); i++) {
       buildAttribute(parser, attr.item(i));
     }
-    if(!MessageML.MESSAGEML_TAG.equals(getMessageMLTag())) {
-      updateBiContext(parser.getBiContext());
-    }
+
     NodeList children = element.getChildNodes();
 
     for (int i = 0; i < children.getLength(); i++) {
       buildNode(parser, children.item(i));
     }
+
+    if(!MessageML.MESSAGEML_TAG.equals(getMessageMLTag())) {
+      updateBiContext(parser.getBiContext());
+    }
   }
 
   /**
-   * Update the BiContext adding information about the MessageML element. If not overridden in the element itself
-   * it is going to create/update a simple BiItem containing the element name and the occurrences.
+   * Update the BiContext adding information about the MessageML element. By default is checking if the element contains
+   * any style or class, to be overridden in every element we want to define additional items.
    */
-  void updateBiContext(BiContext context){
-    context.updateItem(getClass().getSimpleName(), getMessageMLTag());
+  void updateBiContext(BiContext context) {
+    if (getAttribute(STYLE_ATTR) != null) {
+      context.updateItem("StylesCustom");
+    }
+    if (getAttribute(CLASS_ATTR) != null) {
+      computeClassAttributeBi(context);
+    }
+  }
+
+  /**
+   * Class attribute inside an element can contain multiple classes separate by space, this method is going to split
+   * the attribute to extract all different classes if found and fill the corresponding BiItem.
+   *
+   * @param context bi context to be updated
+   */
+  void computeClassAttributeBi(BiContext context) {
```

Comment:
should it be private?

Reply:
yes correct. Fixed in https://github.com/symphonyoss/messageml-utils/pull/266/commits/bedab46386b8a9405936838f80189140567d89d4